### PR TITLE
e2etests-cli: Add PNG test

### LINKED ISSUE #903: cli testing framework
regressions like https://github.com/terrastruct/d2/issues/889 are bad and will continue to happen unless we have tests for the CLI.

### LINKED ISSUE #918: test PNG exports
we'll need to test PNG exports, which means we'll need playwright to work on CI. is it doable on the Github provided runners or do we need to switch to our own then?

motivation:
recently had a regression with PNG exports: https://github.com/terrastruct/d2/issues/911

### PR DESCRIPTION
**Pull request by nhooyr**
Closes #903
Closes #918

### CODE REVIEW

**Review by alixander — COMMENTED** [review]
can u add a test with flag just to demo too? like different `pad` or something. 
maybe a multi-board one too?

i wonder why this png looks blurry. are you on latest d2 or release d2?

**Comment by nhooyr** [comment]
> i wonder why this png looks blurry. are you on latest d2 or release d2?

This uses my xmain PR to run d2 in memory as though it's running externally. So it's the same commit which was just rebased on master.

**Comment by nhooyr** [comment]
Oh haha looks like I didn't run with `TA=1` after rebasing.

**Comment by alixander** [comment]
actually the blurry png was last release, so it should've sorted itself. weiiiird. let me try pulling and running and seeing if i get a different PNG than u after u run with TA=1

**Comment by nhooyr** [comment]
PNGs aren't blurry for me after I ran with `TA=1`.

File: e2etests-cli/main_test.go [hunk 1]
```diff
@@ -2,20 +2,42 @@ package e2etests_cli
...
+				assert.Success(t, err)
+				png := readFile(t, dir, "hello-world.png")
+				assert.Testdata(t, ".png", png)
+			},
+		},
+		{
+			name: "hello_world_png_pad",
+			run: func(t *testing.T, ctx context.Context, dir string, env *xos.Env) {
+				writeFile(t, dir, "hello-world.d2", `x -> y`)
+				err := runTestMain(t, ctx, dir, env, "-pad 4000", "hello-world.d2", "hello-world.png")
```
Comment (nhooyr) [inline e2etests-cli/main_test.go:36]: Wait a second this isn't working.

**Comment by nhooyr** [comment]
Fixed ✅

File: e2etests-cli/main_test.go (hunk 1 above)
Comment (alixander) [inline e2etests-cli/main_test.go]: it's `--pad`

weirdly it throws an error when value too high 

Screen Shot 2023-03-02 at 8 09 38 PM

Comment (nhooyr) [reply e2etests-cli/main_test.go]: Yup already fixed.

Comment (nhooyr) [reply e2etests-cli/main_test.go]: That is super weird though.

Comment (nhooyr) [reply e2etests-cli/main_test.go]: Maybe the PNG gets too big?

**Review by alixander — COMMENTED** [review]
Screen Shot 2023-03-02 at 8 28 26 PM

looks like they do fail in CI =(

**Comment by alixander** [comment]
maybe u need to install dependencies differently like this guy: https://github.com/terrastruct/d2/issues/744#issuecomment-1446641870

File: e2etests-cli/main_test.go [hunk 2]
```diff
@@ -2,20 +2,86 @@ package e2etests_cli
...
+				err := runTestMain(t, ctx, dir, env, "--sketch", "hello-world.d2", "hello-world.png")
+				assert.Success(t, err)
+				png := readFile(t, dir, "hello-world.png")
+				assert.Testdata(t, ".png", png)
+			},
+		},
+		{
+			name: "multiboard/life",
+			run: func(t *testing.T, ctx context.Context, dir string, env *xos.Env) {
+				writeFile(t, dir, "life/index.d2", `x -> y
```
Comment (alixander) [inline e2etests-cli/main_test.go:56]: why does this need to be specified like that? does `life.d2` not work?

Comment (nhooyr) [reply e2etests-cli/main_test.go:56]: It doesn't, I added this just now to test if you point to a directory, `d2` will automatically pick up the `index.d2`. I can add another multiboard to test what you're asking.

**Review by alixander — COMMENTED** [review]
Screen Shot 2023-03-02 at 8 41 39 PM

sadly the `sketch` is still a problem. let's skip/disable that test until the issue is found/fixed

**Comment by nhooyr** [comment]
Do you want me to disable d2renderer sketch tests too then? You can just push a commit after to make it work on arm64 and I'll remember to not touch.

**Comment by alixander** [comment]
> Do you want me to disable d2renderer sketch tests too then? You can just push a commit after to make it work on arm64 and I'll remember to not touch.

no those don't compare diffs remember, they just output it so that the visual changes are shown in PRs

**Review by alixander — APPROVED** [review]
so good to have these, covering a huge blind spot for d2. i'm sure many bugs will be found as we populate these tests. 💯

**Pull request merged**